doc/CMakeLists.txt

```diff
@@ -3,8 +3,7 @@
 cmake_minimum_required(VERSION 3.20.0)
 project(Zephyr-Kernel-Doc LANGUAGES)
 
-set(NO_BOILERPLATE TRUE)
-find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE} ..)
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE} .. COMPONENTS doc)
```

Comment:
missing

python
west
extensions
zephyr_module

Reply:
it can.
I'm fine either way.

I just felt it would look clean to those working with the docs to see a `doc` component instead of the individual modules used.

@gmarull do you have any preference on:
```
 find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE} .. COMPONENTS doc) 
```
vs
```
 find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE} .. COMPONENTS extensions python west root zephyr_module) 
```

Reply:
IMHO since it's bad practice for

some_c_file.c

to do

#include "some_c_file.h"

and then have that file include all the deps of the file, then it should also be considered bad practice to have this doc.cmake file.

Reply:
CMake modules are not directly comparable to C, although I agree it's a good principle to have in mind.

As said earlier I don't have strong preference, but I would like to allow other to chip in with opinions.